docs/okeps/okep-5296-bgp.md

```diff
@@ -0,0 +1,663 @@
+
+### Datapath Performance
+
+In cases where throughput is a priority, using the underlay directly can eliminate the need for tunnel encapsulation,
+and thus reducing the overhead and byte size of each packet. This allows for greater throughput.
+
+### Multi-homing, Link Redundancy, Fast Convergence
+
+BGP can use multi-homing with ECMP routing in order to provide layer 3 failover. When a link goes down, BGP can reroute
+via a different path. This functionality can be coupled with BFD in order to provide fast failover. A typical use case
+is in a spine and leaf topology, where a node has multiple NICs for redundancy that connect to different BGP routers in
+topology. In this case if a link goes down to one BGP peer or the BGP peer fails, BFD can detect the outage on the order
+of milliseconds to hundreds of milliseconds and quickly assist BGP in purging the routes to that failed peer. This then
+causes fast failover and route convergence to the alternate peer.
+
+## User-Stories/Use-Cases
+
+* As a user I want to be able to leverage my existing BGP network to dynamically learn routes to pods in my Kubernetes
+  cluster.
+* As a user, rather than having to maintain routes with NM State manually in each Kubernetes node, as well as being
+  constrained to using local gateway mode for respecting user defined routes; I want to use BGP so that I can dynamically
+  advertise egress routes for the Kubernetes pod traffic in either gateway mode.
+* As a user where maximum throughput is a priority, I want to reduce packet overhead by not having to encapsulate
+  traffic with Geneve.
+* As an egress IP user, I do not want to have to restrict my nodes to the same layer 2 segment and prefer
+  to use a pure routing implementation to handle advertising egress IP movement across nodes.
+
+## Proposed Solution
+
+OVN-Kubernetes will leverage other projects that already exist to enable BGP in Linux. FRR will be used as the BGP
+speaker, as well as provide support for other protocols like BFD. For FRR configuration, the
+MetalLB project has already started an API to be able to configure FRR:
+[https://github.com/metallb/frr-k8s](https://github.com/metallb/frr-k8s). While some of the configuration support for
+FRR may be directly exposed by FRR-K8S API, it may also be the case that some intermediary CRD provided by
+OVN-Kubernetes is required to integrate OVN-Kubernetes networking concepts into FRR.
+
+Functionally, FRR will handle advertising and importing routes and configuring those inside a Linux VRF. OVN-Kubernetes
+will be responsible for listening on netlink and configuring logical routers in OVN with routes learned
+by FRR. OVN-Kubernetes will manage FRR configuration through the FRR-K8S API in order to advertise routes outside of the
+node onto the provider network.
+
+There should be no changes required in FRR. FRR-K8S may need extended APIs to cover the OVN-Kubernetes use cases
+proposed in this enhancement. OVN will require no changes.
```

Comment:
do we need to reconsider the future where OVN has support for learning BGP routers from the host? so whatever  copying is done by OVN-Kubernetes today _can be done by OVN_?
but OK I think we can keep the enhancement as close as possible to the original to be honest and not deviate much so let this be..